source/server/server.cc

```diff
@@ -34,6 +34,7 @@
 #include "common/router/rds_impl.h"
 #include "common/runtime/runtime_impl.h"
 #include "common/singleton/manager_impl.h"
+#include "common/ssl/isfips.h"
```

Comment:
seems you're adding another hook point in Envoy::Server to Common::Ssl, which will make disabling BoringSSL harder, thoughts?

Reply:
We can also simply guard this code with `envoy_select_boringssl()` and `#ifdef BORINGSSL_FIPS`, so that the dependency would be added only in BoringSSL FIPS builds, and hardcode `server.fips` to `0` otherwise...

I've added the code doing that, let me know if it works for you.

Reply:
IMO we should revert the stat entirely so can this be a moot point?

Reply:
I'm confused, yesterday you've said:

> LGTM. Definitely have it output as part of the admin endpoint also. I agree we want runtime access.

Reply:
Sorry I meant via admin endpoint (probably /server_info) vs. a stat. I think that would be better? Doesn't the version already get printed as part of /server_info so we would see it?

Reply:
Ah, sorry, I misunderstood it. I've added it as a stat, since there is `server.version` already, but yeah, `/server_info` might be more appropriate... and yes, it's already part of the `version` string there, which might be good enough, so I'll just remove the stat.

Reply:
@mattklein123 I think this also fits with the general problem of versioning and feature flag detection, will think about this when putting together a proposal today.

Reply:
I'm talking about how to announce the FIPS support of the Envoy node; rather than cramming this into version strings, this fits with a more general feature announcement mechanism.

Reply:
OK yes that would be great.

Reply:
https://github.com/envoyproxy/envoy/issues/5405